modules/navier_stokes/src/fvkernels/INSFVTurbulentDiffusion.C

```diff
@@ -0,0 +1,186 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "INSFVTurbulentDiffusion.h"
+
+registerMooseObject("NavierStokesApp", INSFVTurbulentDiffusion);
+
+InputParameters
+INSFVTurbulentDiffusion::validParams()
+{
+  InputParameters params = FVDiffusion::validParams();
+  params.addClassDescription(
+      "Computes residual for the turbulent scaled diffusion operator for finite volume method.");
+  params.addParam<MooseFunctorName>("scaling_coef", 1.0, "diffusion coefficient");
+  params.set<unsigned short>("ghost_layers") = 2;
+  params.addParam<std::vector<BoundaryName>>(
+      "walls", {}, "Boundaries that correspond to solid walls.");
+
+  return params;
+}
+
+INSFVTurbulentDiffusion::INSFVTurbulentDiffusion(const InputParameters & params)
+  : FVDiffusion(params),
+    _scaling_coef(getFunctor<ADReal>("scaling_coef")),
+    _wall_boundary_names(getParam<std::vector<BoundaryName>>("walls"))
+{
+}
+
+void
+INSFVTurbulentDiffusion::initialSetup()
+{
+  FVDiffusion::initialSetup();
+  for (const auto & elem : _fe_problem.mesh().getMesh().element_ptr_range())
+  {
+    auto wall_bounded = false;
+    for (unsigned int i_side = 0; i_side < elem->n_sides(); ++i_side)
+    {
+      const std::vector<BoundaryID> side_bnds = _subproblem.mesh().getBoundaryIDs(elem, i_side);
+      for (const BoundaryName & name : _wall_boundary_names)
+      {
+        BoundaryID wall_id = _subproblem.mesh().getBoundaryID(name);
+        for (BoundaryID side_id : side_bnds)
+        {
+          if (side_id == wall_id)
+            wall_bounded = true;
+        }
+      }
+    }
+    _wall_bounded[elem] = wall_bounded;
+  }
```

Comment:
Don't you have a function for this already in `NavierStokesMethods`?

Reply:
Changed. Thanks